evm/vm/state.py

```diff
@@ -195,34 +187,6 @@ def get_block_class(cls) -> Type['BaseBlock']:
     #
     # Execution
     #
-
-    def do_call(self, transaction):
-
-        _transaction = transaction
-
-        _transaction.v = DEFAULT_DO_CALL_V
-        _transaction.s = DEFAULT_DO_CALL_S
-        _transaction.r = DEFAULT_DO_CALL_R
-
-        snapshot = self.snapshot()
-        try:
-            if not hasattr(_transaction, "get_sender"):
-                _transaction.get_sender = \
-                    lambda: to_bytes(hexstr=DEFAULT_DO_CALL_SENDER)
-                _transaction.sender = to_bytes(hexstr=DEFAULT_DO_CALL_SENDER)
-
-            # set the account balance of the sender to an arbitrary large
-            # amount to ensure they have the necessary funds to pay for the
-            # transaction.
-            self.account_db.set_balance(transaction.sender, UINT_256_MAX // 2)
```

Comment:
This looks like it was removed and not added back in to ensure gas estimations work from unfunded accounts.  Did I miss something?

Reply:
Rather than pre-funding the address, Im setting the gasPrice to 0, per @carver suggestion.  It will have different behavior than other implementations.

For example go-ethereum will let you check the gas estimation on a transaction that could never occur, like a value transfer from an account with insufficient funds.
This implementation will throw an error in such a case, but in other cases will still let you estimate the gas without having sufficient funds to pay the gas.